### ISSUE TITLE
CanisterStatus - missing controllers

### ISSUE THREAD

**Issue by infu** [dfinity/agent-js#644]
**Describe the bug**
According to documentation, the following code should give module_hash and controllers

documentation: https://internetcomputer.org/docs/current/references/ic-interface-spec/#state-tree-canister-information
```
import { CanisterStatus, HttpAgent } from "@dfinity/agent";

const requestCanisterStatus = (canister_id) => {
  const agent = new HttpAgent({ host: `https://ic0.app` });
  //if (local) await agent.fetchRootKey();
  return CanisterStatus.request({
    canister_id,
    agent,
    paths: ["controllers", "module_hash"],
  });
};
```

It currently only provides module_hash and controllers is empty array.

I thought the API is not giving it for privacy reasons, but I was told IC should be providing it. So it's probably AgentJS bug?
<img width="786" alt="image" src="https://user-images.githubusercontent.com/24810/195096379-d014bd57-2ccd-4fe3-bcbb-85dbe7cdee23.png">

Using  @dfinity/agent 0.14.0

**Comment by ocluf** [comment]
I'm experiencing the same issue. Added the black hole canister as a controller but it does not show up. 
Something even weirder is that when I removed the blackhole canister again with

`dfx canister --network=ic update-settings --remove-controller e3mmv-5qaaa-aaaah-aadma-cai migeh-kqaaa-aaaai-aatrq-cai`

the principal from my dfx didn't show up anymore as well even though it still was a controller

**Comment by krpeacock** [comment]
fixed in #799
